api/src/services/sample-location-service.ts

```diff
@@ -64,13 +64,24 @@ export class SampleLocationService extends DBService {
    */
   async insertSampleLocations(sampleLocations: PostSampleLocations): Promise<SampleLocationRecord[]> {
     const methodService = new SampleMethodService(this.connection);
-
+    const determineName = (geometry: Feature<Geometry | GeometryCollection, Properties>) => {
+      const nameKey = Object.keys(geometry.properties ?? {}).find(
+        (key) => key.toLowerCase() === 'name' || key.toLowerCase() === 'label'
+      );
+      return nameKey && geometry.properties ? geometry.properties[nameKey].substring(0, 50) : '';
+    };
+    const determineDesc = (geometry: Feature<Geometry | GeometryCollection, Properties>) => {
+      const descKey = Object.keys(geometry.properties ?? {}).find(
+        (key) => key.toLowerCase() === 'desc' || key.toLowerCase() === 'description'
```

Comment:
Shapefiles can't have property names longer than 10 characters, so it will never be `'description'`.

Some random internet sites seem to think `'des'` and `'descr'` are other common shorthand versions, so could add those if you want. Ultimately the bios will just have to be told what to use tho, so... :/

Reply:
I added those for good measure. Though it's as you say, ultimately we can only provide so many options.